### PR TITLE
Finish the test suite

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: test/src/index.ts [hunk 1]
```diff
@@ -1019,19 +1397,58 @@ describe('phosphor-widget', () => {
...
       it('should propagate `before-hide` to all non-hidden descendants', () => {
-
+        var bottom0 = new LogWidget();
+        bottom0.hidden = true;
+        var bottom1 = new LogWidget();
+        var middle = new LogWidget([bottom0, bottom1]);
+        var top = new LogWidget([middle]);
+        attachWidget(top, document.body);
+        top.hidden = true;
+        // XXX: should the hidden descendant get notified?
```
Comment (blink1073) [inline test/src/index.ts]: I'd like to draw your attention here.

Comment (sccolbert) [inline test/src/index.ts]: No, the hidden descendant (and any of its descendents) should not be notified. They are already not-visible.

Comment (sccolbert) [inline test/src/index.ts]: That is, after-hide/before-show relate to changes in the `isVisible` state of the widget, not the `hidden` state of the widget.

**Comment by sccolbert** [comment]
Cheers!

**Pull request merged**